lib/api_base_rest/cache/cache.dart

```diff
@@ -0,0 +1,35 @@
+import "dart:convert";
+import "dart:typed_data";
+
+import "package:flutter_cache_manager/flutter_cache_manager.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "../../shared_api_clients/sks_api_client.dart";
+import "cache_manager.dart";
+
+Future<T> getAndCacheData<T>(
+  String url,
+  Ref ref,
+  CacheManager cacheManager,
+  T Function(Map<String, dynamic> json) fromJson,
+  bool Function() condition,
+) async {
+  final cachedFile = await cacheManager.getFileFromCache(url);
+  if (cachedFile != null && condition()) {
+    final cachedData = await cachedFile.file.readAsString();
+    final data = fromJson(
+      jsonDecode(cachedData) as Map<String, dynamic>,
+    );
+    return data;
+  }
+  final dio = ref.read(sksClientProvider);
```

Comment:
we can prolly accept Dio as argument too to make it reusable for other URLs too.

Reply:
Or actually we can maybe make one Dio singleton and always accept full url instead?

Reply:
what you think is better

Reply:
also `ref.watch` is prolly more proper in this situation than `ref.read`

Reply:
I guess we can migrate to using single Dio client and accept full urls and then the functoin could have signuture like this (assuming its an extension on ref)

```dart
getAndCacheData(
  String url,
  int ttlDays,
  T Function(Map<String, dynamic> json) fromJson,
  bool Function() extraValidityCheck,
)
```